vulnerabilities/importers/apache_kafka.py

```diff
@@ -7,120 +7,177 @@
-        self.version_api = GitHubTagsAPI()
-        asyncio.run(self.version_api.load_api(["apache/kafka"]))
+    def advisory_data(self):
+        advisory_page = self.fetch_advisory_page(self)
 
-    def updated_advisories(self):
-        advisory_page = self.fetch_advisory_page()
-        self.set_api()
         parsed_data = self.to_advisory(advisory_page)
-        return self.batch_advisories(parsed_data)
+        return parsed_data
 
     def to_advisory(self, advisory_page):
         advisories = []
+
         advisory_page = BeautifulSoup(advisory_page, features="lxml")
         cve_section_beginnings = advisory_page.find_all("h2")
         for cve_section_beginning in cve_section_beginnings:
-            cve_id = cve_section_beginning.text.split("\n")[0]
+            # This sometimes includes text that follows the CVE on the same line -- sometimes there is a carriage return, sometimes there is not
+            # cve_id = cve_section_beginning.text.split("\n")[0]
+            # This is superior, gets only the cve id and no following text.
+            cve_id = cve_section_beginning.get("id")
+
             cve_description_paragraph = cve_section_beginning.find_next_sibling("p")
+
+            stripped_cve_description_paragraph = str(cve_description_paragraph.get_text())
+            stripped_cve_description_paragraph = stripped_cve_description_paragraph.replace(
+                "\n", ""
+            )
+            stripped_cve_description_paragraph = " ".join(
+                stripped_cve_description_paragraph.split()
+            )
+
             cve_data_table = cve_section_beginning.find_next_sibling("table")
             cve_data_table_rows = cve_data_table.find_all("tr")
             affected_versions_row = cve_data_table_rows[0]
             fixed_versions_row = cve_data_table_rows[1]
-            affected_version_ranges = to_version_ranges(
-                affected_versions_row.find_all("td")[1].text
-            )
-            fixed_version_ranges = to_version_ranges(fixed_versions_row.find_all("td")[1].text)
-
-            fixed_packages = [
-                PackageURL(type="apache", name="kafka", version=version)
-                for version in self.version_api.get("apache/kafka").valid_versions
-                if any(
-                    [
-                        MavenVersion(version) in version_range
-                        for version_range in fixed_version_ranges
-                    ]
-                )
-            ]
 
-            affected_packages = [
-                PackageURL(type="apache", name="kafka", version=version)
-                for version in self.version_api.get("apache/kafka").valid_versions
-                if any(
-                    [
-                        MavenVersion(version) in version_range
-                        for version_range in affected_version_ranges
-                    ]
-                )
+            affected_versions_string = affected_versions_row.find_all("td")[1].text
```

Comment:
Why would need a `_string` suffix? keep this simpler.

Reply:
Agreed and simplified.